src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -955,4 +997,120 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
     const userSessionEndingTime = DeadlineExtensionHelper.getOngoingUserFeedbackSessionEndingTimestamp(feedbackSession);
     return (userSessionEndingTime - Date.now()) < Milliseconds.IN_FIFTEEN_MINUTES;
   }
+
+  /**
+   * Filter questions that we are submitting for intended recipient
+   * when grouped session view is toggled and save the responses after.
+   */
+  saveResponsesForSelectedRecipientQuestions(recipientId: string,
+    questionSubmissionForms: QuestionSubmissionFormModel[]): void {
+    const questionsToRecipient: Set<number> = this.recipientQuestionMap.get(recipientId) || new Set();
+    const recipientQSForms = questionSubmissionForms
+      .filter((questionSubmissionFormModel: QuestionSubmissionFormModel) =>
+          questionsToRecipient.has(questionSubmissionFormModel.questionNumber));
+
+    this.saveFeedbackResponses(recipientQSForms);
+  }
+
+  private addQuestionForRecipient(recipientId: string, questionId: any): void {
+    if (this.recipientQuestionMap.has(recipientId)) {
+      this.recipientQuestionMap.get(recipientId)!.add(questionId);
+    } else {
+      const feedbackQuestionIds: Set<any> = new Set<any>();
+      feedbackQuestionIds.add(questionId);
+      this.recipientQuestionMap.set(recipientId, feedbackQuestionIds);
+    }
+  }
+
+  toggleViewChange(selectedView: SessionView): void {
+    if (selectedView === this.currentSelectedSessionView) {
+      return;
+    }
+
+    if (selectedView === SessionView.DEFAULT) {
+      this.currentSelectedSessionView = SessionView.DEFAULT;
+    } else if (selectedView === SessionView.GROUP_RECIPIENTS) {
+      this.currentSelectedSessionView = SessionView.GROUP_RECIPIENTS;
+      this.groupQuestionsByRecipient();
+    }
+  }
+
+  /**
+   * Group questions by recipients in {@code GROUP_RECIPIENTS} view.
+   */
+  groupQuestionsByRecipient(): void {
+    if (!this.hasLoadedAllRecipients) {
```

Comment:
nit: could convert to guard clause here. if (has loaded) return;

This way the rest of the block has one less indent, easier to read.